src/test/scala/io/citrine/lolo/DataGenerator.scala

```diff
@@ -0,0 +1,113 @@
+package io.citrine.lolo
+
+import breeze.stats.variance
+import io.citrine.lolo.linear.LinearRegressionLearner
+import io.citrine.lolo.stats.functions.Friedman
+import io.citrine.random.Random
+
+object DataGenerator {
```

Comment:
This stuff is pretty much all copied over from `TestUtils`.

The entire purpose of the `TrainingData` interface is to allow us to generate these rows of training data and perform the binning operation on them while maintaining strong typing on the labels. The previous `TestUtils.binTrainingData` function did not allow for this and was incompatible with the `Learner[T]` interface.